test/e2e/gc_e2e_test.go

```diff
@@ -284,4 +282,91 @@ var _ = Describe("Garbage collection for dependent resources", func() {
 
 	})
 
+	Context("Given a configmap and secret associated with a CSV they should be deleted when the CSV is deleted", func() {
+		const (
+			configmapName = "dependentcfg"
+			secretName    = "dependentsecret"
+		)
+
+		var (
+			owner        v1alpha1.ClusterServiceVersion
+			fetchedOwner *v1alpha1.ClusterServiceVersion
+			propagation  metav1.DeletionPropagation
+			options      metav1.DeleteOptions
+			configMap    *corev1.ConfigMap
+			secret       *corev1.Secret
+		)
+
+		BeforeEach(func() {
+			var err error
+			owner = newCSV("owner", testNamespace, "", semver.MustParse("0.0.0"), nil, nil, newNginxInstallStrategy("test", nil, nil))
+			// create all owners
+			fetchedOwner, err = operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Create(context.TODO(), &owner, metav1.CreateOptions{})
+			Expect(err).NotTo(HaveOccurred())
+
+			// wait for csv install to complete
+			Eventually(func() bool {
+				csv, _ := operatorClient.OperatorsV1alpha1().ClusterServiceVersions(testNamespace).Get(context.TODO(), fetchedOwner.GetName(), metav1.GetOptions{})
+				return csv.Status.Phase == v1alpha1.CSVPhaseSucceeded
+			}).Should(BeTrue())
+
+			cfg := &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      configmapName,
+					Namespace: testNamespace,
+				},
+				Data: map[string]string{"hello": "there"},
+			}
+
+			s := &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      secretName,
+					Namespace: testNamespace,
+				},
+				Data: map[string][]byte{"name": []byte("Alice")},
+			}
+
+			// set ownerrefs to objects
+			ownerutil.AddOwner(cfg, fetchedOwner, true, false)
+			ownerutil.AddOwner(s, fetchedOwner, true, false)
```

Comment:
Nit: This tests ensures that secrets that are owned by the CSV are GC'd when the CSV is deleted, but it does not ensure that the Secrets have the ownerrefs. Could come back to bite us.

Reply:
That is a good point, I'm wondering if there is a need for an additional follow-up test, I'll look into our existing tests and see if something like this exists. I imagine it would be a csv e2e test.